### ISSUE TITLE
Class property initialization flaws fails if same class name is used

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#958]
**A\\X::class**

```php
namespace A;
class X { }
```

**B\\X::class**

```php
namespace B;
class X { }
```

Note that we have two classes called `X`, but one from `A` namespace and the another from `B` namespace.

**ParentClass::class**

```php
use A\X;

class ParentClass {
    public $properties = [ X::class ];
}
```

**ChildClass::class**

```php
use B\X;

class ParentClass extends ParentClass {
    public $properties = [ X::class ];
}
```

Now the `ParentClass` import the `A\X`, while `ChildClass` impot `B\X`. 

The problem here is that this inspection consider the "text" that is the same: `X::class`, but it should consider the full classpath from both cases to consider it identical.

The best solution is if the SDK turn available some kind of method to get a `toString()`-like method that could return it with full classpath (_namespaced_), then compare the two sides. 

Else, it should identify that it is an array, then check if have some class element on it. Easy mode: in this case, disable inspection. Best mode: check class by class to identify equality (maybe via some intersect method).

**Issue in a pack:** [issue-958.zip](https://github.com/kalessil/phpinspectionsea/files/2135000/issue-958.zip)

**Comment by kalessil** [comment]
Confirmed, will provide fix in the next release.

**Comment by kalessil** [comment]
Fixed in #1039!
